### PR TITLE
Task/dss90 improve job docs

### PR DESCRIPTION
**Pull request by sarinasinick**
Adding some minor changes to improve the docs for the following: 

* added missing docstrings for several `bundle` methods
* adding more detailed job definition input information, specifically surrounding possible `output` types - in combination with https://github.com/dataiku/dss-doc/pull/538 to help clarify job output syntax and available types 
* improved some of the formatting so that the docstrings display as expected on the doc page

### CODE REVIEW

**Comment by sarinasinick** [comment]
@FChataigner would you mind reviewing some docstring updates to fill out the docs more on some functions that are undocumented?

File: dataikuapi/dss/project.py [hunk 1]
```diff
@@ -1035,17 +1034,34 @@ def get_api_service(self, service_id):
...
                 "/projects/%s/bundles/exported/%s" % (self.project_key, bundle_id))
 
     def get_exported_bundle_archive_stream(self, bundle_id):
         """
         Download a bundle archive that can be deployed in a DSS automation Node, as a binary stream.
-        Warning: this stream will monopolize the DSSClient until closed.
+        
+        .. warning::
+            
+            this stream will monopolize the DSSClient until closed, so the stream must be closed after use or used within a ``with`` statement. For example:
```
Comment (FChataigner) [inline dataikuapi/dss/project.py:1058]: I'd drop the "this stream will monopolize the DSSClient until closed" and just say what needs to be done

**Review by FChataigner — CHANGES_REQUESTED** [review]

File: dataikuapi/dss/project.py (hunk 1 above)
Comment (sarinasinick) [reply dataikuapi/dss/project.py]: 👍  updated

**Review by FChataigner — APPROVED** [review]
LGTM

**Pull request merged**